feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package telemetry_port_speed_test implements tests that cover port-speed related
+// telemetry variables.
+package telemetry_port_speed_test
+
+import (
+	"fmt"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/netutil"
+	"github.com/openconfig/ondatra/telemetry"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Test cases:
+//   - Validate that the port speed reported in telemetry is the expected port speed.
+//   - Turn port down at ATE, validate that operational status of the port is reported as down.
+//   - Connect N ports between ATE and DUT configured as part of a LACP bundle.
+//     Validate that the effective speed of the LAG is reported as N*port speed.
+//   - Disable each port at ATE and determine that the effective speed is reduced by the expected amount.
+//   - Turn ports sequentially up at the ATE, and determine that the effective speed is increased as expected.
+//
+// Topology:
+//
+//	dut:port1 <--> ate:port1
+//	dut:portN <--> ate:portN
+const (
+	plen4 = 30
+	plen6 = 126
+)
+
+var (
+	dutIPs = attrs.Attributes{
+		Name:    "dutip",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::5",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	ateIPs = attrs.Attributes{
+		Name:    "ateip",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::6",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+)
+
+const (
+	lagTypeLACP   = telemetry.IfAggregate_AggregationType_LACP
+	lagTypeSTATIC = telemetry.IfAggregate_AggregationType_STATIC
+	minLink       = 1
+)
+
+type testCase struct {
+	minlinks uint16
+	lagType  telemetry.E_IfAggregate_AggregationType
+
+	dut *ondatra.DUTDevice
+	ate *ondatra.ATEDevice
+	top *ondatra.ATETopology
+
+	dutPorts []*ondatra.Port
+	atePorts []*ondatra.Port
+	aggID    string
+}
+
+func (*testCase) configDUT(i *telemetry.Interface, a *attrs.Attributes) {
+	i.Description = ygot.String(a.Desc)
+	if *deviations.InterfaceEnabled {
```

Comment:
I don't think `deviations.InterfaceEnabled` should cover this - having `/interfaces/interface/config/enabled=true` seems like something that should be supported on all devices. I believe the deviation and discussion was about whether we do this specifically for IPv4. @dplore - PTAL.

Reply:
That is correct, `deviations.InterfaceEnabled` is only intended to cover the ipv4 and ipv6 subinterface leaves.  This doesn't break this test because it is also required by openconfig that configured interfaces are enabled , but makes it slightly more permissive than necessary.